File: backend/src/main/java/com/projeto/mamaecontente/repository/ArtesanatoRepository.java
package com.projeto.mamaecontente.repository;

import com.projeto.mamaecontente.model.Artesanato;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ArtesanatoRepository extends JpaRepository<Artesanato, Long> {
}

File: mobile/app/src/main/java/com/projeto/mamaecontente/data/repository/ArtesanatoRepository.kt
package com.projeto.mamaecontente.data.repository

import com.projeto.mamaecontente.domain.model.ArtesanatoItem
import kotlinx.coroutines.flow.Flow

interface ArtesanatoRepository {

    fun getArtesanatos(): Flow<List<ArtesanatoItem>>

    fun getArtesanatoById(id: Int): Flow<ArtesanatoItem?>

    suspend fun addArtesanato(artesanato: ArtesanatoItem)

    suspend fun updateArtesanato(artesanato: ArtesanatoItem)

    suspend fun deleteArtesanato(id: Int)
}
